Fatal Exception: UninitializedPropertyAccessException

**Describe the bug**
Noticed an issue where if the order filter dialog is visible and the app sent to the background, the app will crash when coming back from the foreground if the activity is being rebuilt

**To Reproduce**
Steps to reproduce the behavior:
1. Turn on `DONT KEEP ACTIVITIES` in the device Developer Options
2. Open the Order list filter dialog
3. Navigate to device home
4. Bring app back to the foreground
5. App crashes with the following error

```
 E  FATAL EXCEPTION: main
 E  Process: com.woocommerce.android.dev, PID: 13012
 E  java.lang.RuntimeException: Unable to start activity ComponentInfo{com.woocommerce.android.dev/com.woocommerce.android.ui.main.MainActivity}: kotlin.UninitializedPropertyAccessException: latei
    nit property orderStatusOptions has not been initialized
 E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
 E      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
 E      at android.app.ActivityThread.-wrap11(Unknown Source:0)
 E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
 E      at android.os.Handler.dispatchMessage(Handler.java:106)
 E      at android.os.Looper.loop(Looper.java:164)
 E      at android.app.ActivityThread.main(ActivityThread.java:6494)
 E      at java.lang.reflect.Method.invoke(Native Method)
 E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
 E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
 E  Caused by: kotlin.UninitializedPropertyAccessException: lateinit property orderStatusOptions has not been initialized
 E      at com.woocommerce.android.ui.orders.OrderStatusFilterDialog.getOrderStatusOptions(OrderStatusFilterDialog.kt:49)
 E      at com.woocommerce.android.ui.orders.OrderStatusFilterDialog$filterMap$2.invoke(OrderStatusFilterDialog.kt:43)
 E      at com.woocommerce.android.ui.orders.OrderStatusFilterDialog$filterMap$2.invoke(OrderStatusFilterDialog.kt:18)
 E      at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
 E      at com.woocommerce.android.ui.orders.OrderStatusFilterDialog.getFilterMap(Unknown Source:7)
 E      at com.woocommerce.android.ui.orders.OrderStatusFilterDialog.getCurrentOrderStatusIndex(OrderStatusFilterDialog.kt:52)
 E      at com.woocommerce.android.ui.orders.OrderStatusFilterDialog.onCreateDialog(OrderStatusFilterDialog.kt:56)
 E      at android.support.v4.app.DialogFragment.onGetLayoutInflater(DialogFragment.java:328)
 E      at android.support.v4.app.Fragment.performGetLayoutInflater(Fragment.java:1226)
 E      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1428)
 E      at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1759)
 E      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1827)
 E      at android.support.v4.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3244)
 E      at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:3200)
 E      at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:195)
 E      at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:597)
 E      at android.support.v7.app.AppCompatActivity.onStart(AppCompatActivity.java:177)
 E      at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1334)
 E      at android.app.Activity.performStart(Activity.java:7029)
 E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2741)
 E      ... 9 more
```

--- a/RELEASE-NOTES.txt
+++ b/RELEASE-NOTES.txt
@@ -14,6 +14,7 @@ Bugfixes
 - Fixed rare crash after undoing a review moderation
 - Fixed cases where the sales totals for a time period don't match what Calypso and wp-admin report
 - Fixed rare crash loading product images in top earners 
+- Fixed a rare crash when recreating the order filter during activity restore (low memory situations)
 
 Improvements
 - Custom order status labels are now supported! Instead of just displaying the order status slug, the custom order

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderListFragment.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderListFragment.kt
@@ -65,6 +65,7 @@ class OrderListFragment : TopLevelFragment(), OrderListContract.View, OrderStatu
     @Inject lateinit var selectedSite: SelectedSite
 
     private lateinit var ordersDividerDecoration: DividerItemDecoration
+    private var orderFilterDialog: OrderStatusFilterDialog? = null
 
     override var isRefreshPending = true // If true, the fragment will refresh its orders when its visible
     override var isRefreshing: Boolean
@@ -212,6 +213,14 @@ class OrderListFragment : TopLevelFragment(), OrderListContract.View, OrderStatu
             }
         }
         return view
+    }
+
+    override fun onPause() {
+        super.onPause()
+
+        // If the order filter dialog is visible, close it
+        orderFilterDialog?.dismiss()
+        orderFilterDialog = null
     }
 
     override fun onResume() {
@@ -479,8 +488,9 @@ class OrderListFragment : TopLevelFragment(), OrderListContract.View, OrderStatu
     // region Filtering
     private fun showFilterDialog() {
         val orderStatusOptions = presenter.getOrderStatusOptions()
-        OrderStatusFilterDialog.newInstance(orderStatusOptions, orderStatusFilter, listener = this)
-                .show(fragmentManager, OrderStatusFilterDialog.TAG)
+        orderFilterDialog = OrderStatusFilterDialog
+                .newInstance(orderStatusOptions, orderStatusFilter, listener = this)
+                .also { it.show(fragmentManager, OrderStatusFilterDialog.TAG) }
     }
 
     override fun onFilterSelected(orderStatus: String?) {

--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderStatusFilterDialog.kt
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderStatusFilterDialog.kt
@@ -72,7 +72,7 @@ class OrderStatusFilterDialog : DialogFragment() {
                         // If 'All' is selected filter, pass null to signal a filterless refresh
                         listener?.onFilterSelected(selectedFilter.takeUnless { it == ALL_FILTER_ID })
                     }
-                    dialog.cancel()
+                    dialog.dismiss()
                 }.create()
     }
 

